### ISSUE TITLE
Introduction of Few-Shot Examples to Improve Accuracy of Terminal Copilot

### ISSUE THREAD

**Issue by 2mawi2** [Methexis-Inc/terminal-copilot#27]
Since GPT is a few-shot learner model see: [Language Models are Few-Shot Learners](https://proceedings.neurips.cc/paper/2020/file/1457c0d6bfcb4967418bfb8ac142f64a-Paper.pdf) we can significantly improve the model performance by providing examples instead of only providing context in a zero shot approach.

**Comment by JoelKronander** [comment]
Maybe, this could be related to your other idea of having a router to select the correct few-shot prompt to send, given the context. E.g. is the user asking about git commands, we could add a few related few-shot prompts for git commands etc?

**Comment by 2mawi2** [comment]
If we want todo this properly we would require some high level component that generates our prompt. From there we could tuning it into different directions. I would argue that less context could be more for the fine-tuning. We could for example have a generic amount of few shot examples Z and then include only a subset of Z in order to finetune the prompt in the specific direction (e.g. git).
